Programs/_testembed.c

```diff
@@ -2323,6 +2323,187 @@ test_create_module_from_initfunc(void)
     return Py_RunMain();
 }
 
+/// Multi-phase initialization package & submodule ///
+
+int
+mp_pkg_exec(PyObject *mod)
+{
+    // make this a namespace package
+    PyObject *path_list = PyList_New(0);   // empty list = namespace package
+    if (!path_list) {
+        return -1;
+    }
+    if (PyModule_AddObject(mod, "__path__", path_list) < 0) {
+        Py_DECREF(path_list);
+        return -1;
+    }
+    if (PyModule_AddStringConstant(mod, "mp_pkg_exec_slot_ran", "yes") < 0) {
+        Py_DECREF(path_list);
```

Comment:
Subtle refcounting bug in the error case.

[PyModule_Add](https://docs.python.org/3/c-api/module.html#c.PyModule_Add) can make things easier:

```suggestion
    // empty list = namespace package
    if (PyModule_Add(mod, "__path__", PyList_New(0)) < 0) {
        return -1;
    }
    if (PyModule_AddStringConstant(mod, "mp_pkg_exec_slot_ran", "yes") < 0) {
```